tdrs-backend/tdpservice/email/tasks.py

```diff
@@ -69,3 +73,77 @@ def email_admin_num_access_requests():
                                    subject,
                                    email_context,
                                    )
+
+
+@shared_task
+def send_data_submission_reminder(due_date, reporting_period, fiscal_quarter):
+    """Send all Data Analysts a reminder to submit if they have not already."""
+    def get_fiscal_year(calendar_year, fiscal_quarter):
+        return calendar_year - 1 if fiscal_quarter == 'Q1' else calendar_year
+
+    now = datetime.now()
+    fiscal_year = get_fiscal_year(now.year, fiscal_quarter)
+
+    all_locations = STT.objects.all()
+
+    #
+    reminder_locations = []
+    for loc in all_locations:
+        submitted_sections = DataFile.objects.all().filter(
+            stt=loc,
+            year=fiscal_year,
+            quarter=fiscal_quarter,
+            # version=??
+        )
+        required_sections = loc.filenames.keys()
+
+        submitted_all_sections = True
+        for s in required_sections:
```

Comment:
Question. Can we avoid this loop by just checking the length of the `required_sections` against the length of the `submitted_sections`? I think all you would need to add (if we can do this) is `.distinct('section')`.

Reply:
thank you, this was a good suggestion. i started to implement and thought of one possible edge case

i guess it's technically possible (for now) to submit a section you're not required to submit (i.e., section 4 isn't always required but you can submit for it anyway). i could still clean this up to where it doesn't require a query in every iteration of the loop, but i'm thinking the loop is still necessary. thoughts?